OpenSim/Simulation/Model/Geometry.h

```diff
@@ -611,7 +615,12 @@ class OSIMSIMULATION_API Mesh : public Geometry
     /// Method to map Mesh to Array of SimTK::DecorativeGeometry.
     void implementCreateDecorativeGeometry(
         SimTK::Array_<SimTK::DecorativeGeometry>& decoGeoms) const override;
-
+private:
+    // We cache the DecorativeMeshFile and a flag indicating if we tried to
+    // load the mesh from file so we don't try loading from disk every frame.
+    // These are mutable since they're not part of the public interface.
+    mutable std::shared_ptr<SimTK::DecorativeMeshFile> cachedMesh;
+    mutable bool triedToLoad;
```

Comment:
This (as well as some of my other comments) might become irrelevant based on @aseth1's feedback.